src/util/verify-chromium-port.js

```diff
@@ -0,0 +1,52 @@
+import { createServer } from 'node:http';
+
+import { UnusablePortError } from '../errors.js';
+
+/**
+ * Determine if a port is available
+ * @param {number} port Port to test
+ * */
+async function isPortAvailable(port) {
+  return new Promise((resolve) => {
+    const server = createServer();
+    server.once('listening', () => {
+      server.close(() => resolve(true));
+    });
+    server.once('error', () => resolve(false));
+    server.listen(port);
+  });
+}
+
+/**
+ * Validate that requested port is a valid port
+ * @param {number} port Debugging port
+ * @returns {boolean}
```

Comment:
This `isPortValid` helper` and the exported `isPortValid` helper document a boolean return value, but throw errors when the port is invalid. This is not documented, and strange.

To avoid the strange "true / false / Error` tristate, you could consider dropping the boolean return values and just have error vs non-Error.

```js
// Raises an error if the port is invalid for any reason
await ensureValidPort(port);
```

Another option is to have a boolean return value (port available or not) and throw one error (`PortNotAvailableError`), with the diagnostic information exported via the `log` object ([example of createLogger](https://github.com/mozilla/web-ext/blob/1c015188bc215cdbd327bb5d2213727f178d6231/src/extension-runners/firefox-android.js#L17-L20)).